lib/core.js

```diff
@@ -307,6 +307,15 @@ interface Iterable<T> {
     @@iterator(): Iterator<T>;
 }
 
+/* Generators */
+interface Generator<Y,R,N> {
+    @@iterator(): Iterator<Y>;
+    next(value?: N): IteratorResult<Y|R>;
+}
+
```

Comment:
So here's what I'm trying to do. If you `yield *` to a generator, your PR accurately tries to match the delegating generator's `R` and `N` to the delegated generator's `R` and `N`. This is right and correct and awesome.

If you're `yield *`'ing to something that is NOT a generator but IS an `Iterable`, then you're not doing anything with the delegating generator's `R` and `N`. This is fine if the `Iterable` is something like an array, since `R` is `void` and `N` is `void`. However, I can conceive of a custom `Iterable` who's `Iterator` is something fancy, like a generator.

So given the example above

    yield *({ [Symbol.iterator]: function *() { return yield 4; } })

If you call `next('batman')`, this `yield *` expression should evaluate to `batman`. (The example in babel: https://goo.gl/1Xq5VV)

I think it's possible to capture this idea, by creating `$Iterator` and `$Iterable` which keep track of the `N` and `R` types too. I'm not possible it will work, but it would be cool!